mailbox/memory/src/main/resources/META-INF/spring/mailbox-memory.xml

```diff
@@ -1,57 +1,58 @@
-        <constructor-arg index="6" ref="storeMailboxAnnotationManager" />
-        <constructor-arg index="7" ref="storeRightManager" />
-        <constructor-arg index="8" ref="quotaComponents" />
-        <constructor-arg index="9" ref="indexer" />
-    </bean>
-
-    <bean id ="memory-subscriptionManager" class="org.apache.james.mailbox.store.StoreSubscriptionManager">
-        <constructor-arg index="0" ref="memory-sessionMapperFactory"/>
-    </bean>
-
-    <bean id="memory-sessionMapperFactory" class="org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory" />
-
-    <alias name="jvm-locker" alias="memory-locker"/>
-
-    <bean id="memory-mailbox-id-deserializer" class="org.apache.james.mailbox.inmemory.InMemoryMailboxIdDeserializer"/>
-
-</beans>
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.    
+-->
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+          http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <bean id="memory-mailboxIdFactory" class="org.apache.james.mailbox.inmemory.InMemoryId.Factory" />
+
+    <!-- 
+      Mailbox Memory
+     -->
+    <!-- WARNING: Memory does not persist the mailbox. -->
+    <!--          Information will be lost after jvm restart. -->
+
+    <bean id="memory-mailboxManager" class="org.apache.james.mailbox.inmemory.InMemoryMailboxManager">
+        <constructor-arg index="0" ref="memory-sessionMapperFactory"/>
+        <constructor-arg index="1" ref="sessionProvider"/>
+        <constructor-arg index="2" ref="memory-locker"/>
+        <constructor-arg index="3" ref="messageParser"/>
+        <constructor-arg index="4" ref="messageIdFactory" />
+        <constructor-arg index="5" ref="event-bus" />
+        <constructor-arg index="6" ref="storeMailboxAnnotationManager" />
+        <constructor-arg index="7" ref="storeRightManager" />
+        <constructor-arg index="8" ref="quotaComponents" />
+        <constructor-arg index="9" ref="indexer" />
+        <constructor-arg index="10"><set/></constructor-arg>
```

Comment:
ref?

Reply:
Duc will spend a litte time to explain this modification for me? It is related to `Spring`?